src/main/protobuf/dwrf_proto.proto

```diff
@@ -159,13 +172,46 @@ message StripeInformation {
     optional uint64 footerLength = 4;
     optional uint64 numberOfRows = 5;
     optional uint64 rawDataSize = 6;
+    // Arbitrary metadata similar for stripe level encryption
+    // Optimization?? only set for run start, and reuse until next run
```

Comment:
@zzhao0 can you clarify the status of this optimization (is it a part of the file format? what are the implications)

Reply:
The optimization basically is same as that in open source ORC

We set keyMetadata once and have subsequent stripes (and the footer) reuse the key if key is not changed. This is part of the format. It's ok to set it for every single stripe and reader won't fail (if implemented correct to handle the runs)